crates/miden-lib/src/account/auth/mod.rs

```diff
@@ -46,36 +46,72 @@ impl From<RpoFalcon512> for AccountComponent {
 /// An [`AccountComponent`] implementing a procedure-ACL based RpoFalcon512 signature scheme for
 /// authentication of transactions.
 ///
-/// This component only requires authentication when any of the specified procedures are called
-/// during the transaction. It stores a list of procedure roots that require authentication, and
-/// the signature verification is only performed if at least one of these procedures was invoked.
+/// This component provides fine-grained authentication control based on three conditions:
+/// 1. **Procedure-based authentication**: Requires authentication when any of the specified trigger
+///    procedures are called during the transaction.
+/// 2. **Output note authentication**: Controls whether creating output notes requires
+///    authentication. Output notes are new notes created by the account and sent to other accounts
+///    (e.g., when transferring assets). When `allow_unauthorized_output_notes` is `false`, any
+///    transaction that creates output notes must be authenticated, ensuring account owners control
+///    when their account sends assets to other accounts.
+/// 3. **Input note authentication**: Controls whether consuming input notes requires
+///    authentication. Input notes are notes that were sent to this account by other accounts (e.g.,
+///    incoming asset transfers). When `allow_unauthorized_input_notes` is `false`, any transaction
+///    that consumes input notes must be authenticated, ensuring account owners control when their
+///    account processes incoming notes.
 ///
-/// It exports the procedure `auth__tx_rpo_falcon512_procedure_acl`, which:
-/// - Checks if any of the specified auth trigger procedures were called during the transaction
-/// - If none were called, authentication is skipped
-/// - If at least one was called, performs standard RpoFalcon512 signature verification
-/// - Always increments the nonce
+/// ## Authentication Logic
 ///
-/// The storage layout is:
+/// Authentication is required if ANY of the following conditions are true:
+/// - Any trigger procedure from the ACL was called
```

Comment:
As you mentioned elsewhere, should we introduce the "ACL" acronym somewhere close to the beginning of this doc section, e.g. "Access Control List (ACL)"